File: library/src/main/java/com/example/ruru/library/builder/SimpleFootBuilder.java
package com.example.ruru.library.builder;

import com.example.ruru.library.viewholder.BaseViewHolder;

public abstract class SimpleFootBuilder implements FootBuilder {

    /**
     * 正常底部布局：不用于分页加载数据时的布局
     *
     * @param holder
     */
    public void onNormal(BaseViewHolder holder) {

    }

    /**
     * 正在加载时底部的布局
     *
     * @param holder
     */
    public void onLoading(BaseViewHolder holder) {

    }

    /**
     * 加载失败时底部的布局
     *
     * @param holder
     */
    public void onLoadingFailure(BaseViewHolder holder) {

    }

    /**
     * 没有数据时底部的布局
     *
     * @param holder
     */
    public void onNoMoreData(BaseViewHolder holder) {

    }


}

File: library/src/main/java/com/example/ruru/library/builder/SimpleFootBuilder.kt
package com.example.ruru.library.builder

import com.example.ruru.library.viewholder.BaseViewHolder

/**
 * Created by lyr on 2019/12/15
 */
abstract class SimpleFootBuilder: FootBuilder {
  /**
   * 正常底部布局：不用于分页加载数据时的布局
   */
  override fun onNormal(holder: BaseViewHolder) {}

  /**
   * 正在加载时底部的布局
   */
  override fun onLoading(holder: BaseViewHolder) {}

  /**
   * 加载失败时底部的布局
   */
  override fun onLoadingFailure(holder: BaseViewHolder) {}

  /**
   * 没有数据时底部的布局
   */
  override fun onNoMoreData(holder: BaseViewHolder) {}
}